src/models/ChatMessage.js

```diff
@@ -0,0 +1,44 @@
+const mongoose = require('mongoose');
+
+const chatMessageSchema = new mongoose.Schema(
+  {
+    albumId: {
+      type: String,
+      required: true,
+    },
+    message: {
+      type: String,
+      required: true,
+    },
+    userId: {
+      type: mongoose.Schema.ObjectId,
+      ref: 'User',
+      required: true,
+    },
+    timestamp: {
+      type: Date,
+      default: Date.now,
+    },
+  },
+  {
+    capped: { size: 102400, max: 1000 }, // max 1000 messages
+  }
+);
+
+//console.log('Schema Created');
+
+// Creating a virtual field "album" that references the Album model
+chatMessageSchema.virtual('album', {
+  ref: 'Album', // Reference to the Album model
+  localField: 'albumId', // Local field for the relationship
+  foreignField: 'spotify_id', // Foreign field for the relationship
+  justOne: true, // Refers to a single album
+});
+
+//console.log('Virtual Field "album" Created');
```

Comment:
Honestly, I think I somehow totally confused myself when I gave the suggestion to add this virtual before, I'm so sorry :sweat: :sweat\_smile: . We can remove this code for now.

So far in our code, we don't actually care about fetching a full Album object from a ChatMessage. The only thing we need in our `chatRoutes` is the ability to search by the value that we got from the API url, which we can do without any additional modification. 

If there's a use case in the future where we need the full album from a `ChatMessage` object, then we can revisit this. Apologies for any confusion.